src/main/scala/org/broadinstitute/dsde/firecloud/model/ElasticSearch.scala

```diff
@@ -33,27 +33,40 @@ trait ESPropertyFields {
     include_in_all = Some(false),
     store = Some(true)
   )
+  // https://www.elastic.co/guide/en/elasticsearch/reference/2.4/search-suggesters-completion.html
+  def completionField = ESInnerField(
+    "completion",
+    analyzer = Option("simple"),
+    search_analyzer = Option("simple")
+  )
   def rawField(`type`:String) = ESInnerField(`type`,
     index = Some("not_analyzed")
   )
 }
 
 // top-level field defs, for facet and non-facet types
-case class ESType(`type`: String, copy_to: String) extends ESPropertyFields
+case class ESType(`type`: String, fields: Option[Map[String,ESInnerField]], copy_to: Option[String] = None ) extends ESPropertyFields
 object ESType extends ESPropertyFields {
-  def apply(`type`: String):ESType = ESType(`type`, ElasticSearch.fieldSuggest)
+  def apply(`type`: String, hasPopulateSuggest: Boolean, hasSearchSuggest: Boolean, isAggregatable: Boolean):ESType =  {
+    val fields: Option[Map[String,ESInnerField]] = (hasPopulateSuggest, isAggregatable) match {
+      case (true, true) => Option(Map("suggest" -> completionField, "raw" -> rawField(`type`)))
+      case (true, false) => Option(Map("suggest" -> completionField))
+      case (false, true) => Option(Map("raw" -> rawField(`type`)))
+      case (false, false) => None
```

Comment:
yes, we wanted to reduce the multiple case classes to one. i'm definitely open to suggestions for how to improve.